web/models/terms/term.ex

```diff
@@ -31,5 +32,17 @@ defmodule CoursePlanner.Terms.Term do
     |> validate_required([:name, :start_date, :end_date, :status])
     |> cast_embed(:holidays)
     |> Statuses.update_status_timestamp(EntityStatus)
+    |> validate_date_range()
   end
+
+  defp validate_date_range(%{errors: []} = changeset) do
+    st = changeset |> Changeset.get_field(:start_date) |> Date.cast!
```

Comment:
do you really need to cast those dates? it seems to me they were supposed to be casted already